backend/spec/requests/users_spec.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe("Users") do
+  describe "POST /users/" do
+    it "returns user object with token if params are valid" do
+      post("/users/", params: {
+        user: {
+          username: "bhoss_killah_mapagmahal",
+          password: "p@ssw0rd",
+        },
+      })
+      expect(response).to(have_http_status(:created))
+      user = json_body[:data][:user]
+      decoded_token = JsonWebToken.decode(json_body[:data][:token])
+
+      expect(user[:id]).to(eql(decoded_token["user_id"]))
+      expect(json_body[:errors]).to(be_nil)
+    end
+
+    it "returns error if username params already exists" do
+      post("/users/", params: {
+        user: {
+          username: "existing_user",
+          password: "p@ssw0rd",
+        },
+      })
+      post("/users/", params: {
+        user: {
+          username: "existing_user",
+          password: "p@ssw0rd",
+        },
+      })
+      expect(response).to(have_http_status(:unprocessable_entity))
```

Comment:
Good job on the RSpec. This is actually already great, pero if u want some more things to improve you can try the one expectation rule.

https://rspec.rubystyle.guide/
https://devblast.com/b/ruby-testing-with-rspec-one-expectation-per-test

It's a bit old na and not always followed but whenever someone sees this, super easy to read tests mo, they'll thank u sa code reviews haha